libp2p.go

```diff
@@ -0,0 +1,90 @@
+package libp2p
+
+import (
+	"context"
+	"crypto/rand"
+
+	crypto "github.com/libp2p/go-libp2p-crypto"
+	host "github.com/libp2p/go-libp2p-host"
+	pnet "github.com/libp2p/go-libp2p-interface-pnet"
+	metrics "github.com/libp2p/go-libp2p-metrics"
+	peer "github.com/libp2p/go-libp2p-peer"
+	pstore "github.com/libp2p/go-libp2p-peerstore"
+	swarm "github.com/libp2p/go-libp2p-swarm"
+	transport "github.com/libp2p/go-libp2p-transport"
+	bhost "github.com/libp2p/go-libp2p/p2p/host/basic"
+	mux "github.com/libp2p/go-stream-muxer"
+	ma "github.com/multiformats/go-multiaddr"
+	mplex "github.com/whyrusleeping/go-smux-multiplex"
+	msmux "github.com/whyrusleeping/go-smux-multistream"
+	yamux "github.com/whyrusleeping/go-smux-yamux"
+)
+
+type Config struct {
+	Transports  []transport.Transport
+	Muxer       mux.Transport
+	ListenAddrs []ma.Multiaddr
+	PeerKey     crypto.PrivKey
+	Peerstore   pstore.Peerstore
+	Protector   pnet.Protector
+	Reporter    metrics.Reporter
+}
+
+func Construct(ctx context.Context, cfg *Config) (host.Host, error) {
```

Comment:
I wonder if having New(ctx) and NewWithConfig(ctx, cfg) would be nicer. At least for me as a new user it would be more straightforward to understand what those methods do anf how they work